### ISSUE TITLE
to_html()没有encoding参数？造成输出中文显示乱码

### ISSUE THREAD

**Issue by 645187919** [pandas-dev/pandas#28663]
#### Code Sample, a copy-pastable example if possible

```python
# Your code here
data = pd.read_csv("test.csv",encoding="gbk")
print(data)
html = data.to_html(col_space=40,justify="center")
```
#### Problem description
当我用to_html（）方法生成html的时候，中文显示乱码，后来一看该方法竟然没有encoding参数，希望作者能完善了，这样以后会方便很多人哈！
[this should explain **why** the current behaviour is a problem and why the expected output is a better solution.]

**Note**: We receive a lot of issues on our GitHub tracker, so it is very possible that your issue has been posted before. Please check first before submitting so that we do not have to handle and close duplicates!

**Note**: Many problems can be resolved by simply upgrading `pandas` to the latest version. Before submitting, please check if that solution works for you. If possible, you may want to check if `master` addresses this issue, but that is not necessary.

For documentation-related issues, you can check the latest versions of the docs on `master` here:

https://pandas-docs.github.io/pandas-docs-travis/

If the issue has not been resolved there, go ahead and file it in the issue tracker.

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]

</details>

**Comment by simonjayhawkins** [comment]
@645187919 Thanks for the report, however, can you provide a reproducible example. http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports

**Comment by charlesdong1991** [comment]
I think he is say there is no `encoding` argument in `to_html`, and this will cause an issue when exporting contents including Chinese. @simonjayhawkins

**Comment by simonjayhawkins** [comment]
Thanks @charlesdong1991 .

the `to_html` method has no encoding. I assume that it is fairly simple to work-around by passing an open buffer with the correct encoding or not passing a filepath, and writing the return unicode string to file using the appropriate encoding?

the buffer handling code is shared with `to_latex` and `to_string`.

`to_latex` already has an encoding parameter and the shared code can handle encoding.

so it should be straightforward to add encoding to `to_html` ( and `to_string`) for consistency.

https://github.com/pandas-dev/pandas/issues/15008#issuecomment-269638646 "Textual payloads should support the encoding argument"

#28377 has been opened for discussion regarding passing options for the open call.

but I think in this case, for consistency with other IO methods, we could just add the encoding argument directly to the to_html argument list. @TomAugspurger

**Comment by charlesdong1991** [comment]
yeah, agree, a solution to solve this issue could be adding `encoding` to `to_html` in `io/formats/format.py` @simonjayhawkins 
And a separate PR could be making it sharable among those three methods.

**Comment by ailurus1991** [comment]
added at https://github.com/pandas-dev/pandas/pull/28692

**Comment by farziengineer** [comment]
Added at #28951
